### PR TITLE
ben/gre-1161-add-start-messages-to-json-transcript

### PR DESCRIPTION
**Pull request by benjamin-sylvanus**
[ticket](https://linear.app/greymatter/issue/GRE-1161/add-start-messages-to-json-transcript)

Test Calls:
[Outbound](https://app.opencall.ai/console/calls/WzEsICJwdWJsaWMiLCAiY2FsbHMiLCAxMTU5MF0%3D)
[Inbound](https://app.opencall.ai/console/calls/WzEsICJwdWJsaWMiLCAiaW5ib3VuZF9jYWxscyIsIDUzOTUwXQ%3D%3D)

Is this what we were looking for, please listen to the recording and look at the json transcript.

### CODE REVIEW

File: vocode/streaming/streaming_conversation.py [hunk 1]
```diff
@@ -1004,15 +1004,15 @@ async def send_initial_message(self, initial_message: BaseMessage):
...
+        if self.agent.get_agent_config().call_type == CallType.OUTBOUND:
+            self.agent.update_history(
+                "human", self.transcriptions_worker.buffer.to_message()
             )
-        )
-        self.agent_responses_worker.consume_nonblocking(agent_response_event)
+            self.agent.update_history("message.bot", initial_message.text)
 
+        elif self.agent.get_agent_config().call_type == CallType.INBOUND:
+            self.agent.update_history("message.bot", initial_message.text)
```
Comment (benjamin-sylvanus) [inline vocode/streaming/streaming_conversation.py:1015]: Should pass the `initial_message_tracker` to the update history function

File: vocode/streaming/streaming_conversation.py [hunk 2]
```diff
@@ -1004,15 +1004,15 @@ async def send_initial_message(self, initial_message: BaseMessage):
...
-                AgentResponseMessage(message=initial_message),
-                is_interruptible=False,
-                agent_response_tracker=initial_message_tracker,
+        # If it is outbound, update the history with the initial message
+        if self.agent.get_agent_config().call_type == CallType.OUTBOUND:
+            self.agent.update_history(
+                "human", self.transcriptions_worker.buffer.to_message()
             )
-        )
-        self.agent_responses_worker.consume_nonblocking(agent_response_event)
```
Comment (bluegoo192) [inline vocode/streaming/streaming_conversation.py:1014]: it looks like `agent.update_history` does create an `AgentResponseMessage`, but with the loss of this line we are no longer consuming it anywhere. What are the consequences of this?

File: vocode/streaming/streaming_conversation.py (hunk 1 above)
Comment (bluegoo192) [reply vocode/streaming/streaming_conversation.py:1015]: makes sense yeah

**Review by bluegoo192 — CHANGES_REQUESTED** [review]
thanks for doing this, the results look great. I'm just worried about an edge case regression since we're not consuming the start message event anymore, and we should add the message tracker back in

File: vocode/streaming/streaming_conversation.py (hunk 2 above)
Comment (benjamin-sylvanus) [reply vocode/streaming/streaming_conversation.py:1014]: `self.agent_responses_worker.consume_nonblocking(agent_response_event)`
The agent responses worker consumes agent response events. So in the update history function we do the equivalent. 
[ref](https://github.com/greymatter-labs/vocode-python-main/blob/840587e57dd410853a4eb9b22f0800b02ec3d0ac/vocode/streaming/agent/state_agent.py#L426C1-L429C14)

**Comment by benjamin-sylvanus** [comment]
> thanks for doing this, the results look great. I'm just worried about an edge case regression since we're not consuming the start message event anymore, and we should add the message tracker back in

the behavior appears to be the same. I'm not sure we need to add the tracker back in but would be easy enough to do I think. It helps with tracing and I believe initially control of the transcribers mute. I'll take a look later

**Comment by benjamin-sylvanus** [comment]
INBOUND:
trace id inbound: 6a5e8d5bc346f4b222d758e7b0ad0d8b
https://app.opencall.ai/console/calls/WzEsICJwdWJsaWMiLCAiaW5ib3VuZF9jYWxscyIsIDU0NzUxXQ%3D%3D
Screenshot 2024-10-06 at 7 22 50 PM

OUTBOUND:
https://app.opencall.ai/console/calls/WzEsICJwdWJsaWMiLCAiY2FsbHMiLCAxMjE1OF0%3D
Screenshot 2024-10-06 at 7 23 06 PM

trace id outbound: f8a88986f3a0e1cfde610ed234c6bff2

File: vocode/streaming/streaming_conversation.py [hunk 3]
```diff
@@ -1004,15 +1004,16 @@ async def send_initial_message(self, initial_message: BaseMessage):
...
         initial_message_tracker = asyncio.Event()
-        agent_response_event = (
-            self.interruptible_event_factory.create_interruptible_agent_response_event(
-                AgentResponseMessage(message=initial_message),
-                is_interruptible=False,
-                agent_response_tracker=initial_message_tracker,
+        # If it is outbound, update the history with the initial message
+        if self.agent.get_agent_config().call_type == CallType.OUTBOUND:
+            self.agent.update_history(
+                "human", self.transcriptions_worker.buffer.to_message()
```
Comment (bluegoo192) [inline vocode/streaming/streaming_conversation.py:1010]: we don't want to use the tracker for outbound?

**Review by bluegoo192 — COMMENTED** [review]
is it just me, or did it try to say the start message twice in a row on the inbound call?

File: vocode/streaming/streaming_conversation.py (hunk 3 above)
Comment (benjamin-sylvanus) [reply vocode/streaming/streaming_conversation.py:1010]: Ummm this is the human message.

Comment (benjamin-sylvanus) [reply vocode/streaming/streaming_conversation.py:1010]: Outbound we add the current transcription. Then both inbound and outbound update the history on line 1012 for the bot.

**Review by bluegoo192 — APPROVED** [review]

**Comment by benjamin-sylvanus** [comment]
So the behavior Oliver requested is not represented in main. Need to look at this him before moving forward.

**Pull request closed without merging**